src/portfft/committed_descriptor_impl.hpp

```diff
@@ -0,0 +1,1301 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's portFFT
+ *
+ **************************************************************************/
+
+#ifndef PORTFFT_COMMITTED_DESCRIPTOR_IMPL_HPP
+#define PORTFFT_COMMITTED_DESCRIPTOR_IMPL_HPP
+
+#include <sycl/sycl.hpp>
+
+#include <complex>
+#include <cstdint>
+#include <functional>
+#include <numeric>
+#include <vector>
+
+#include "common/exceptions.hpp"
+#include "common/subgroup.hpp"
+#include "defines.hpp"
+#include "enums.hpp"
+#include "specialization_constant.hpp"
+#include "utils.hpp"
+
+namespace portfft {
+
+template <typename Scalar, domain Domain>
+class committed_descriptor_impl;
```

Comment:
Opinion: I think this should be in the detail namespace since we never expect a user of portFFT to use it directly.

Reply:
Good point! Done.